node/src/BaseClient.ts

```diff
@@ -8401,6 +8401,18 @@ export class BaseClient {
             | connection_request.ProtocolVersion
             | undefined;
 
+        // Validate that clientAz is set when using AZ affinity strategies
+        if (
+            options.readFrom === "AZAffinity" ||
+            options.readFrom === "AZAffinityReplicasAndPrimary"
```

Comment:
maybe this would be better
```suggestion
            options.readFrom.includes("AZAffinity")
```

Reply:
I think we can cross that bridge if more AZ strategies get introduced, at the moment the readability is fine and clearly shows the options

Reply:
sure 👍

Reply:
Include is expensive compared to direct compare. In case we expend, it should become an enum/TS type. Don't use expensive functions with no good reason. 
Presently, the current approach seems the cleaner and most efficient.
But you can remove the nested if
```ts
    if ((options.readFrom === "AZAffinity" || "AZAffinityReplicasAndPrimary") && !options.clientAz) throw new ConfigurationError(
        `clientAz must be set when readFrom is set to ${options.readFrom}`,
    );
```

Reply:
agreed, dropped the nesting